src/Components/NewNoteEditor/NewNoteEditor.jsx

```diff
@@ -0,0 +1,109 @@
+                        }}>
+                            <option value="">Select Priority</option>
+                            <option value="HIGH">High</option>
+                            <option value="MODERATE">Moderate</option>
+                            <option value="LOW">Low</option>
+                        </select>
+                    </div>
+                    <button 
+                        className="btn-cancel" 
+                        onClick={(e)=>{
+                           clickHandler(e);
+                    }}>
+                        <i className="fa-solid fa-x"></i>
+                    </button>
+                </div>
+                <input 
+                    type="text" 
+                    placeholder="New Note" 
+                    className="new-note-title" 
+                    value={title}
+                    onChange={(e)=>
+                        noteDispatch({type: "TITLE", payload: e.target.value}) 
+                    }
+                />
+                <textarea 
+                    placeholder="Note it down quick"
+                    onChange={(e)=>
+                        noteDispatch({type: "CONTENT", payload: e.target.value})
+                    }
+                    value={content}
+                />
+                <TagBox/>
+                <div className="action-box">
+                    <i className="fa-solid fa-palette icons clickable-object" 
+                        onClick={()=>
+                            {
+                                optionDispatch({type: "SHOW_PALETTE", payload: !optionState.colorPalette});
+                                if(optionState.createTags) optionDispatch({type: "SHOW_TAGS", payload: false});
+                            }
+                        }
+                    ></i>
+                    <i className="fa-solid fa-box-archive icons clickable-object"></i>
+                    <i className="fa-solid fa-trash-can icons clickable-object"></i>
+                    <i className="fa-solid fa-tags icons clickable-object" 
+                        onClick={()=>
+                            {
+                                optionDispatch({type: "SHOW_TAGS", payload: !optionState.createTags});
+                                if(optionState.colorPalette) optionDispatch({type: "SHOW_PALETTE", payload: false});
+                            }
+                        }
+                    ></i>
+                </div>
+                <ColorPalette showPalette={optionState.colorPalette} />
+                <TagSelect showCreateTag={optionState.createTags} optionDispatch={optionDispatch}/>
+                <div className="display-justify-end">
+                    <button className={`solid-primary btn-save ${title === ""? "solid-disabled" : ""}`}
+                    onClick={(e)=>{
+                       clickHandler(e);
+                       addNotes(newNote);
```

Comment:
Can perform this action only if the content in title and textarea is not empty otherwise alert the user through browser to add some content into the note . Can use trim function to eliminate the edge cases too

Reply:
Didn't think of the function `trim()`. Thanks for pointing this out!